### ISSUE TITLE
loss.backword() doesn't work in vertical federated learning

### ISSUE THREAD

**Issue by MarterLee** [OpenMined/PySyft#2624]
all the examples are about horizontal federated learning. And I don't have any instructions on vertical federated learning.
and I try as follow: 

```python
def train(args,workers,models, device, train_loaders, optimizers, epoch):
	loss_function=nn.MSELoss(size_average=True)
	for i in range(len(workers)):
		models[i].train()
	iter_dataset_Alice=iter(train_loaders[0])
	iter_dataset_Bob = iter(train_loaders[1])
	batch_idx=0
	while(True):
		try:
			data_Alice=next(iter_dataset_Alice)
		except StopIteration:
			break
		features_A,targets_A=data_Alice
		features_A, targets_A=features_A.to(device).send(workers[0]),targets_A.to(device).send(workers[0])
		models[0].send(workers[0])
		optimizers[0].zero_grad()
		output_A= models[0](features_A.float())
		try:
			data_Bob = next(iter_dataset_Bob)
		except StopIteration:
			break
		features_B, targets_B = data_Bob
		features_B, targets_B = features_B.to(device).send(workers[1]), targets_B.to(device).send(workers[1])
		models[1].send(workers[1])
		optimizers[1].zero_grad()
		output_B = models[1](features_B.float())
		output=output_A.get()+output_B.get()		
		output=output.to(device)
		output_A=output.send(workers[0])
		output_B = output.send(workers[1])
		loss_A=loss_function(output_A,targets_A.float())
		#loss_B = loss_function(output_B, targets_B.float())
		loss_A.backward()
		#loss_B.backward()
		optimizers[0].step()
		optimizers[1].step()
		batch_idx=batch_idx+1
		model_A=models[0].get() 
		model_B=models[1].get()
		print(model_A.LR.weight.data)
		print(model_B.LR.weight.data)
```

and the parameters don't update.

why?

the dataset has 130 features ,I split it into two datasets, each has 65 features ,so  add the output of  two models will get the prediction. it's a kind of vertical federated learning.

**Comment by AustinNeverPee** [comment]
Hi, have pysyft already supported the vertical scenarios?

**Comment by AustinNeverPee** [comment]
Hi, has pysyft already supported the vertical scenarios?
